Packs/Active_Directory_Query/Scripts/IAMInitADUser/IAMInitADUser.py

```diff
@@ -1,89 +1,120 @@
+                err = demisto.get(password_outputs[0], 'Contents')
+            else:
+                enable_outputs = demisto.executeCommand("ad-enable-account", ad_create_user_arguments)
+                if is_error(enable_outputs):
+                    err = demisto.get(enable_outputs[0], 'Contents')
+                else:
+                    update_outputs = demisto.executeCommand("ad-update-user", ad_create_user_arguments)
+                    if is_error(update_outputs):
+                        err = demisto.get(update_outputs[0], 'Contents')
+
+        send_mail_outputs = send_email(display_name, username, user_email, err, to_email, password, inc_id)
+
+        success = err is None
+        sent_mail = not is_error(send_mail_outputs)
+        outputs = {
+            'sentMail': sent_mail,
+            'success': success
+        }
 
-def send_email(name, sAMAccountName, user_email, password, to_email):
-    try:
-        if not to_email:
-            return
+        if not success:
+            outputs['errorDetails'] = err
+            readable_output = ERROR_MESSAGE.format(err)
+        if not sent_mail:
+            outputs['sendMailError'] = str(demisto.get(send_mail_outputs[0], 'Contents'))
+            if readable_output:
+                readable_output = '\n\nIn addition, the following error returned from send-mail command:\n'
+            readable_output += outputs['sendMailError']
+
+        if success and sent_mail:
+            readable_output = 'Successfully activated user ' + username + '.'
+
+        result = CommandResults(
+            outputs_prefix='IAM.InitADUser',
+            outputs=outputs,
+            readable_output=readable_output
+        )
+        return_results(result)
 
-        subject = f'{SUBJECT}: {sAMAccountName}'
+    except Exception as e:
+        outputs['success'] = False
+        outputs['errorDetails'] = str(e)
+        result = CommandResults(
+            outputs_prefix='IAM.InitADUser',
+            outputs=outputs,
+            readable_output=outputs['errorDetails']
+        )
+        return_results(result)
+
+
+def send_email(display_name, username, user_email, err, to_email, password, inc_id):
+    if not err:
+        subject = f'IAM - user {display_name} was successfully activated in Active Directory'
         email_body = 'Hello,\n\n' \
                      'The following account has been created in Active Directory:\n\n'
-        if name:
-            email_body += 'Name: ' + name + '\n'
+        if display_name:
+            email_body += 'Name: ' + display_name + '\n'
 
-        email_body += 'sAMAccountName: ' + sAMAccountName + '\n' \
+        email_body += 'username: ' + username + '\n' \
                       'Email: ' + user_email + '\n' \
                       'Password: ' + password + '\n\n' \
                       'Regards,\nIAM Team'
+    else:
+        subject = f'"IAM - Activate User In Active Directory" incident {inc_id} failed with user {display_name}'
```

Comment:
The email subject & email failure subject should be args with default values to allow the users to modify their message/email

Reply:
I added an "email_subject" arg, but for the email body it's more complicated as it contains the account details. FYI, it requires an additional input for the AD User Activation playbook in order to allow configuring the email.

Reply:
You can pass it with the brackets `{}`, no?